.changelog/unreleased/bug-fixes/3206-fix-panic-when-err-is-not-nil.md

```diff
@@ -0,0 +1,2 @@
+- `[log]` Fix panic when err is not nil which can cause panics in the logs
```

Comment:
Can you provide a stack trace of the panic?

Reply:
@melekes there isn't a stack trace with panic, I just opened this PR to prevent it!

Reply:
But this PR is not preventing anything...

Reply:
tbh it's preventing possible issues in the future if someone changes the return type all the other references to that method would work!

but up to you of course 😉